### PR TITLE
added resolution details of analogout

### PR DESCRIPTION
**Pull request by studavekar**

### CODE REVIEW

**Comment by studavekar** [comment]
@SenRamakri @AnotherButler can you please review.

File: docs/reference/api/drivers/AnalogOut.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## AnalogOut
 
-Use the AnalogOut interface to set the voltage of an analog output pin as a floating point number from 0.0 (VSS) to 1.0 (VCC).
+Use the AnalogOut interface to set the voltage of an analog output pin as a floating point number from 0.0 (VSS) to 1.0 (VCC). DAC Resolution depends on the hardware device used, the max resolution supported is VCC/65,536V.
```
Comment (SenRamakri) [inline docs/reference/api/drivers/AnalogOut.md]: @studavekar - "Use the AnalogOut interface to set the output voltage of an analog output pin specified as a percentage or as an unsigned short in the range 0x0000-0xFFFF. Separate API are provided to use percentage or range. Mbed-OS supports a maximum resolution of 1/65535 of the maximum voltage, although actual resolution ,,,, "

**Comment by AnotherButler** [comment]
@SenRamakri Do these changes address your comments?

**Comment by studavekar** [comment]
I feel we should also mention  `DAC Resolution depends on the hardware device`

**Pull request merged**